src/qt/dcclient.cpp

```diff
@@ -126,7 +126,5 @@ wxPaintDCImpl::wxPaintDCImpl( wxDC *owner )
 wxPaintDCImpl::wxPaintDCImpl( wxDC *owner, wxWindow *win )
     : wxWindowDCImpl( owner, win )
 {
-    wxCHECK_RET( m_isWindowPainter,
-                 "wxPaintDC can't be created outside wxEVT_PAINT handler" );
```

Comment:
FWIW, `Qt` won't let you paint anything outside the `paintEvent()` handler. So I don't know how this affects `wxGLCanvas` to be honest!

Reply:
When the assert message dialog appears in Debug, this causes a recursive repaint and recursive assert for some reason.

wxPaintDC is created inside paint handlers in "cube" and "pyramid" samples, but when wxQtGLWidget paint handlers are called, they didn't set the correct flags/data to make the assert disappear.

Reply:
I'll try to look at this a.s.a.p.

Reply:
Thanks for looking at this, I'd prefer not to remove this assert as it can catch a common mistake.

However I also think that it would be nice to make asserts in paint handlers non-fatal: currently they always result in a crash because when the assert message box is shown, another paint event is generated and another assert happens and we crash. And this is not wxQt-specific, exactly the same thing happens in wxMSW and wxGTK too. So ideally we should use something like `wxRecursionGuard` to avoid generating this assert if we're already inside it...

Reply:
The real fix is to remove these overrides:
https://github.com/wxWidgets/wxWidgets/blob/5dac42edc07f6e504ecc9b5a9cbb1c559ca9319c/src/qt/glcanvas.cpp#L37-L40

from `wxQtGLWidget` and just let `wxQtEventSignalHandler` (the base class) do its work as normal

Reply:
If I do that, "pyramid" sample only shows dark green canvas for some reason.

Reply:
Seems to be working after fixing `wxGLContext::SetCurrent`

Reply:
Seems like Qt messes up the clear color and depth test at each paint if these overrides are removed.

Otherwise, `isosurf` sample, when run from executable's dir looks fine if this is added to `wxQtGLWidget` ctor:
```
        setAttribute(Qt::WA_OpaquePaintEvent, true);
        setAttribute(Qt::WA_NoSystemBackground, true);
```

Reply:
With latest commits, all samples work with correct background colors and depth testing.